### ISSUE TITLE
Doc versioning

### ISSUE THREAD

**Issue by Blacksmoke16** [athena-framework/athena#137]
Currently the website only supports a single `latest` version based on the current state of https://github.com/athena-framework/website/tree/master. This makes it hard to manage updates to the website that are not compatible with the latest release as a dedicated branch needs to be created/maintained and not merged until the related code is live.

MKDocs supports a version selector feature that we might be able to leverage in order to have `latest` and `master` versions. Can reference https://github.com/crystal-lang/crystal-book/pull/529 as well. However, the difference in contextual versus API docs needs to  be taken into consideration. As such, there are a few possible scenarios that need to be handled:

1. Pushing a change to contextual docs specific to an unreleased version
2. Pushing a change to contextual docs valid for both unreleased and latest versions
3. Pushing a breaking change to the api docs of a specific component
4. Pushing a backwards compatible change to the api docs of a specific component
5. Releasing a new version of a component

From here these should trigger various combinations of updating docs:

1. Update `master` docs
2. Update `master` and `latest` docs
3. Update `master` API docs for that one component and `framework`
4. Update `master` API docs for that one component and `framework`
5. Update `master` and `latest` docs for that one component and `framework`

Which essentially boils down into two workflows:

1. Handle pushes to contextual docs
2. Handle pushes and releases on API components

The former could be handled via a `push` and a `release`/`manual` GHA workflows. I.e. on changes to contextual docs, update the `master` version, then either when you tag the website or manually trigger an action, update the `latest` version. Alternatively, could have two branches and just merge master into latest when the latest docs need updated. 

On the API component side of things, that would need to run on `push` to any component and when a release is tagged. which could be handled via webhooks possibly I.e. drop `shard.lock` from VC, on push use an override file that fetches master branch and uses that to build and push master version. Then when a release is tagged, use default `shard.yml` to fetch and install latest release and push to latest version.

**Comment by Blacksmoke16** [comment]
I been thinking on this again and have some new thoughts:

* Do the docs really need versioning if only the latest version of each is supported? Probably not.
* Cloudflare pages supports branch based subdomains and scheduled builds such that `nightly.athenaframework.org` could be built nightly using `branch: master` on each component. This would allow viewing docs for unreleased versions, leaving `athenaframework.org` to the current stable releases.
* CF pages also supports preview builds of specific commits
* Having dedicated subdomains for each component is doable, but not really feasible. Would need separate `mkdocs` projects, branches, and inner-component linking would break

Taking into these points into consideration, our 5 original scenarios would be addressed via:

1. Push change to `nightly` branch, leaves `master` unchanged. Users can view nightly docs if working with commit pinned component/framework version
2. Push change to `nightly` branch, also cherrypick it to `master` (or vice versa).
3. Push change to `nightly` branch, leaves `master` unchanged. Docs for that specific component will be viewable, but related components may be out of sync. However I think it's fair to assume if a breaking change happens in a specific component, the other components would be updated to be compatible. Thus allowing `nightly` branch to build, serve as a WIP branch w/o affecting the validity of `master`
4. Push change to `nightly` branch. `nightly` branch will get the changes first and will be applied to `master` via the next scenario
5. New tag makes `master` and `nightly` in-sync at this point.

So in short unless anyone can think of something obvious I'm missing, I think it makes sense to migrate the site to Cloudflare Pages with dedicated branches for latest and nightly releases.
